engines/scalac/src/main/scala/scala/macros/internal/engines/scalac/semantic/Abstracts.scala

```diff
@@ -163,8 +163,9 @@ trait Abstracts extends scala.macros.semantic.Mirrors { self: Universe =>
     }
 
     implicit class XtensionToGType(tpe: Type) {
-      def toGType: g.Type = tpe match {
+      def toGType(implicit m: Mirror): g.Type = tpe match {
         case gtpt: g.TypeTree => gtpt.tpe
+        case tname: Name => Symbol(tname.value)(m).tpe
         case _ => ???
       }
```

Comment:
@xeno-by I did not fix your first comment since it is not clear where I should call .toType in SyntaxAnalyzer. I tried on 306 line but it always complained